src/historywork/DownloadVerifyTxResultsWork.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "historywork/DownloadVerifyTxResultsWork.h"
+#include "history/FileTransferInfo.h"
+#include "history/HistoryManager.h"
+#include "historywork/GetAndUnzipRemoteFileWork.h"
+#include "historywork/Progress.h"
+#include "historywork/VerifyBucketWork.h"
+#include "historywork/VerifyTxResultsWork.h"
+#include "util/format.h"
+#include "work/WorkSequence.h"
+
+#include <medida/meter.h>
+#include <medida/metrics_registry.h>
+
+namespace stellar
+{
+
+DownloadVerifyTxResultsWork::DownloadVerifyTxResultsWork(
+    Application& app, CheckpointRange range, TmpDir const& downloadDir)
+    : BatchWork(app, "download-verify-txresults")
+    , mDownloadDir(downloadDir)
+    , mRange(range)
+    , mCurrCheckpoint(range.first())
+    , mVerifyTxResultsSuccess(app.getMetrics().NewMeter(
+          {"history", "verify-tx-results", "success"}, "event"))
+    , mVerifyTxResultsFailure(app.getMetrics().NewMeter(
+          {"history", "verify-tx-results", "failure"}, "event"))
+{
+}
+
+std::string
+DownloadVerifyTxResultsWork::getStatus() const
+{
+    if (!isDone() && !isAborting())
```

Comment:
I see there are two kind of this condition used for exactly the same thing in code:

1. `if (getState() == BasicWork::State::WORK_RUNNING)`
2. `if (!isDone() && !isAborting())`

Those give two results that are slightly different.
First one is true for internal states of `RUNNING`, `PENDING` and `ABORTING`, and second one for 
`PENDING`, `RUNNING`, `WAITING`, `RETRYING`. I think we should decide on one behavior there, or even change `BasicWork::getStatus` to call new virtual method `getRunningStatus` when `InternalState::RUNNING`.

Reply:
Good idea, I think it should be the second, to correctly handle cases where work is aborting.